src/iosrv/init.c

```diff
@@ -256,7 +256,73 @@ dss_topo_init()
 }
 
 static int
-server_init()
+abt_max_num_xstreams(void)
+{
+	char   *env;
+
+	env = getenv("ABT_MAX_NUM_XSTREAMS");
+	if (env == NULL)
+		env = getenv("ABT_ENV_MAX_NUM_XSTREAMS");
+	if (env != NULL)
+		return atoi(env);
+	return 0;
+}
+
+static int
+set_abt_max_num_xstreams(int n)
+{
+	char   *name = "ABT_MAX_NUM_XSTREAMS";
+	char   *value;
+	int	rc;
+
+	D_ASSERTF(n > 0, "%d\n", n);
+	D_ASPRINTF(value, "%d", n);
```

Comment:
Ah D_ASPRINTF == D_ALLOC_SPRINTF ?  they why not use the whole name. Sorry this is not this patch issue.

Reply:
ah, there is asprintf

Reply:
Yeah, I didn't know asprintf too, until a couple of years ago I learnt from Xuezhao or Johann.